src/control/lib/hardware/fabric.go

```diff
@@ -946,6 +946,13 @@ func (s *FabricScanner) Scan(ctx context.Context, providers ...string) (*FabricI
 	if s == nil {
 		return nil, errors.New("FabricScanner is nil")
 	}
+	provs := []string{}
+	for _, prov := range providers {
+		if prov != "" {
```

Comment:
This was addressing an issue with the hardware/fabric.go scan function which passes in an empty string when called from daos_server network scan. I didn't want to make any more invasive changes than I needed to.
Commenting out this additional change results in unit test failure:
```
--- FAIL: TestDaosServer_Network_Commands_JSON (0.00s)
    --- FAIL: TestDaosServer_Network_Commands_JSON/Scan_network;_JSON;_nothing_found (0.00s)
        network_test.go:76: unexpected stdout (-want, +got):
              (
                """
                {
                  "response": null,
            -     "error": "get local fabric interfaces: no fabric interfaces found",
            +     "error": "get local fabric interfaces: no fabric interfaces found with provider \"\"",
                  "status": -1025
```

Reply:
Ah, I see. Can you add a unit test in lib/hardware/fabric_test.go for this change as well, please?

Reply:
Done, changed one of the error messages as an easy option to avoid false matches in the unit test when only specifying a substring of the error message. Felt than rewriting all the tests to check the entire error string.